presto-native-execution/presto_cpp/main/operators/BroadcastFactory.cpp

```diff
@@ -146,10 +163,16 @@ velox::BufferPtr BroadcastFileReader::next() {
 
   auto readFile = fileSystem_->openFileForRead(
       broadcastFileInfos_[readfileIndex_].filePath_);
-  auto buffer = AlignedBuffer::allocate<char>(readFile->size(), pool_, 0);
-  readFile->pread(0, readFile->size(), buffer->asMutable<char>());
   ++readfileIndex_;
   ++numFiles_;
+
+  // Empty broadcast file.
+  if (readFile->size() <= 0) {
```

Comment:
Shouldn't think be a check? It is not possible to have an empty file here, right?

Reply:
correct, removed this check